docs/modules/ROOT/pages/servlet/configuration/java.adoc

```diff
@@ -250,6 +250,65 @@ public class MultiHttpSecurityConfig {
+
+[source,java]
+----
+@Configuration
+@EnableWebSecurity
+public class CustomSecurityFilterChainConfig {
+
+    @Bean                                                                   <1>
+    public UserDetailsService userDetailsService() {
+        InMemoryUserDetailsManager manager = new InMemoryUserDetailsManager();
+        manager.createUser(User.withDefaultPasswordEncoder().username("user").password("password").roles("ADMIN").build());
+        manager.createUser(User.withDefaultPasswordEncoder().username("admin").password("password").roles("ADMIN").build());
+        return manager;
+    }
+
+    @Bean
+    @Order(1)                                                               <2>
+    public SecurityFilterChain apiFilterChain(HttpSecurity http) throws Exception {
+        http
+                .securityMatcher(RequestMatcher.regexMatchers("/api/**"))   <3>
+                .authorizeHttpRequests(authorize -> authorize
+                        .anyRequest().hasRole("ADMIN")
+                )
+                .httpBasic(withDefaults());
+        return http.build();
+    }
+
+    @Bean
+    @Order(2)                                                                <4>
+    public SecurityFilterChain imagesFilterChain(HttpSecurity http) throws Exception {
+        http
+                .securityMatcher(RequestMatcher.regexMatchers("/images/**")) <5>
+                .authorizeHttpRequests(authorize -> authorize
+                        .anyRequest().hasRole("ADMIN")
+                )
+                .httpBasic(withDefaults());
+        return http.build();
+    }
+
+    @Bean                                                                    <6>
+    public SecurityFilterChain defaultFilterChain(HttpSecurity http) throws Exception {
+        http
+                .authorizeHttpRequests(authorize -> authorize
+                        .anyRequest().authenticated()
+                )
+                .formLogin(withDefaults());
+        return http.build();
+    }
+}
+----
+<1> Begin by configuring authentication settings.
+<2> Define a SecurityFilterChain instance with @Order(1) to handle requests with URLs starting with "/api/". This chain will have the highest priority.
+<3> Specify that the http.securityMatcher applies only to "/api/**" URLs.
+<4> Next, create another SecurityFilterChain instance with @Order(2) to handle requests with URLs starting with "/images/". This chain will be considered second.
+<5> Indicate that the http.securityMatcher applies only to "/images/**" URLs.
+<6> Lastly, create an additional SecurityFilterChain instance without an @Order annotation. This configuration will handle requests not covered by "/api/" or "/images/" URLs and will be processed last (no @Order defaults to last).
```

Comment:
This new example does not seem to provide any new information compared to the previous (existing) example. It simply adds one new pattern with the same authorization rule (`hasRole("ADMIN")`). The new example also doesn't seem to make sense in a real world situation either, as I wouldn't expect an `ADMIN` user to provide credentials via basic authentication in order to access images.

In order to truly improve the documentation, I feel that we need to provide a distinct example of an alternate use case. The new text content we add should correspond directly to the new example.

Would you be able to update this PR to provide a different example and corresponding text content?

Reply:
@sjohnr, I have implemented the changes as requested. Could you please review this merge request again?